source/grid/grid_generator.cc

```diff
@@ -7214,31 +7312,46 @@ namespace GridGenerator
                   cell_data.vertices[i] =
                     local_vertex_indices[index_vertices[i]];
                   cell_data.material_id = material_or_boundary_id;
+                  cell_data.manifold_id = manifold_id; // set cell-manifold id
                 }
               cells.push_back(cell_data);
             }
           else if (dim == 2 && struct_dim == 1) // an edge of a triangle
             {
               CellData<1> boundary_line(2);
               boundary_line.boundary_id = material_or_boundary_id;
+              boundary_line.manifold_id = manifold_id;
               for (unsigned int i = 0; i < index_vertices.size(); ++i)
                 {
                   boundary_line.vertices[i] =
                     local_vertex_indices[index_vertices[i]];
                 }
               subcell_data.boundary_lines.push_back(boundary_line);
             }
-          else if (dim == 3 && struct_dim == 2) // a face of a thetrahedron
+          else if (dim == 3 && struct_dim == 2) // a face of a tetrahedron
             {
               CellData<2> boundary_quad(3);
-              boundary_quad.boundary_id = material_or_boundary_id;
+              boundary_quad.material_id = material_or_boundary_id;
+              boundary_quad.manifold_id = manifold_id;
               for (unsigned int i = 0; i < index_vertices.size(); ++i)
                 {
                   boundary_quad.vertices[i] =
                     local_vertex_indices[index_vertices[i]];
                 }
               subcell_data.boundary_quads.push_back(boundary_quad);
             }
+          else if (dim == 3 && struct_dim == 1) // an edge of a tetrahedron
+            {
+              CellData<1> boundary_line(2);
+              // boundary_line.boundary_id = material_or_boundary_id;
+              boundary_line.manifold_id = manifold_id;
+              for (unsigned int i = 0; i < index_vertices.size(); ++i)
+                {
+                  boundary_line.vertices[i] =
+                    local_vertex_indices[index_vertices[i]];
```

Comment:
This will be a bug if `index_vertices.size()>2`. That should of course never happen, but it would still be good to put a
```
Assert (index_vertices.size() == 2, ExcInternalError());
```
at the top of this block (line 7345, before the declaration of `boundary_line`).

Reply:
In fact, such an assertion would be useful in all of the other blocks above as well.